docs/src/main/asciidoc/application-data-caching-guide.adoc

```diff
@@ -0,0 +1,474 @@
+annotated with `@CacheKey`, or all the arguments annotated with `@CacheKey` otherwise. If a value is found in the cache, it
+is returned and the annotated method is never actually executed. If no value is found, the annotated method is invoked and the
+returned value is stored in the cache using the computed key.
+
+This annotation can't be used on methods returning `void` and requires at least one method argument to compute the key.
+
+=== @CacheStore
+
+When a method annotated with `@CacheStore` is invoked, Quarkus will cache its return value at the end of the invocation. The
+caching is done using a cache key computed from all the method arguments if none of them is annotated with `@CacheKey`, or all
+the arguments annotated with `@CacheKey` otherwise. If a value is already present in the cache for the computed key, it is
+replaced.
+
+This annotation can't be used on methods returning `void` and requires at least one method argument to compute the
+
+=== @CacheInvalidate
+
+When a method annotated with `@CacheInvalidate` is invoked, Quarkus will try to remove an existing entry from the cache. This
+removal is done using a cache key computed from all the method arguments if none of them is annotated with `@CacheKey`, or all
+the arguments annotated with `@CacheKey` otherwise. If the key does not identify any cache entry, nothing will happen.
+
+This annotation requires at least one method argument to compute the key.
+
+[TIP]
+====
+If the `@CacheLoad`, `@CacheStore` or `@CacheInvalidate` annotations are used on a method with no parameters, a unique default
+cache key derived from the cache name will be generated and used.
+====
+
+=== @CacheInvalidateAll
+
+When a method annotated with `@CacheInvalidateAll` is invoked, Quarkus will remove all entries from the cache.
+
+[TIP]
+====
+Each of the `@CacheLoad`, `@CacheStore`, `@CacheInvalidate` and `@CacheInvalidateAll` annotations takes an optional `cacheName`
+parameter which can be used to specify the name of the cache. This is especially useful when the same cache must be accessed
+from several classes. If the cache name is not specified, the fully qualified name of the class where the annotated method is
+declared will be used as the default cache name.
+====
+
+=== @CacheKey
+
+When a method argument is annotated with `@CacheKey`, it is identified as a part of a cache key during an invocation of a
+method annotated with `@CacheLoad`, `@CacheStore` or `@CacheInvalidate`.
+
+This annotation is optional and should only be used when some of the method arguments are NOT part of the cache key.
+
+== Configuring the underlying caching provider
+
+For now, https://github.com/ben-manes/caffeine[Caffeine] is the only available local caching provider in Quarkus. We plan to
+add more providers soon.
+
+=== Caffeine configuration properties
+
+Caffeine is a high performance, near optimal caching library based on Java 8.
```

Comment:
I don't think the `based on Java 8` brings anything to the plate, especially because if feels a bit outdated whereas Caffeine is state of the art.

Reply:
This entire sentence is directly taken from the Caffeine [readme](https://github.com/ben-manes/caffeine/blob/master/README.md) on GH. I'll remove the Java 8 part.

Reply:
I would bet it was written when it was fancy to be based on Java 8. But Quarkus is coming from the future :)

Reply:
Java 8 reference removed.